### PR TITLE
Bump kapp-controller's package to 0.30.1

### LINKED ISSUE #1755: Bump kapp-controller to 0.30.1
(This is used to request new product features)

**Describe the feature request**
Bump kapp-controller to 0.30.1

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

### What this PR does / why we need it
* Bump kapp-controller to 0.30.1
* Add write access to /etc/pki/tls/certs/ca-bundle.crt so kapp-controller will have no issue in writing custom certificates into it.

### Which issue(s) this PR fixes

Fixes #1755 

### Describe testing done for PR
* Local testing
* Downstream test pipeline is successful

### Release note

```release-note
Bump kapp-controller to 0.30.1 and package template bug fix
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#! schema.yaml
```
Comment (blc1996) [inline pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:1]: Wondering if it's desired to include the schema file

**Comment by alfredthenarwhal** [comment]
> @blc1996:  /test
Commit: 0fbfe5d

Tests failed! Build no: 1569

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Review by danniel1205 — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**